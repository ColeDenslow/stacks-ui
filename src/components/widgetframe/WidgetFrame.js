import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import './WidgetFrame.css';

function WidgetFrame(props) {
  const { className, children, scoped, ...otherProps } = props;
  const classNames = cn([
    'stx-widget-frame',
    { 'stx-widget-frame--scoped': scoped },
    className,
  ]);

  return (
    <div
      {...otherProps}
      className={classNames}
    >
      {children}
    </div>
  );
}

WidgetFrame.defaults = {};

WidgetFrame.propTypes = {
  /** Optional className for the WidgetFrame component */
  className: PropTypes.string,
  /** Child elements */
  children: PropTypes.node,
  /** Applies a scoped class so that reset styles can be applied */
  scoped: PropTypes.bool,
};

export default WidgetFrame;
