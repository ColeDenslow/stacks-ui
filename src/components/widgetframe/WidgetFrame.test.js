import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import WidgetFrame from './WidgetFrame';

describe('WidgetFrame', () => {
  test('WidgetFrame renders with basic props', () => {
    const output = mount(<WidgetFrame
      className='the-class'
    >
      <div>Hello world!</div>
    </WidgetFrame>);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('WidgetFrame renders with scoped prop', () => {
    const output = mount(<WidgetFrame
      className='the-class'
      scoped
    >
      <div>Hello world!</div>
    </WidgetFrame>);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('WidgetFrame renders other props', () => {
    const output = mount(<WidgetFrame
      className='the-class'
      scoped
      data-other-prop='other prop value'
    >
      <div>Hello world!</div>
    </WidgetFrame>);

    expect(toJson(output)).toMatchSnapshot();
  });
});
