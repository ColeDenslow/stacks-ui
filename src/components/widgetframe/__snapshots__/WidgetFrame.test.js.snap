// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetFrame WidgetFrame renders other props 1`] = `
<WidgetFrame
  className="the-class"
  data-other-prop="other prop value"
  scoped={true}
>
  <div
    className="stx-widget-frame stx-widget-frame--scoped the-class"
    data-other-prop="other prop value"
  >
    <div>
      Hello world!
    </div>
  </div>
</WidgetFrame>
`;

exports[`WidgetFrame WidgetFrame renders with basic props 1`] = `
<WidgetFrame
  className="the-class"
>
  <div
    className="stx-widget-frame the-class"
  >
    <div>
      Hello world!
    </div>
  </div>
</WidgetFrame>
`;

exports[`WidgetFrame WidgetFrame renders with scoped prop 1`] = `
<WidgetFrame
  className="the-class"
  scoped={true}
>
  <div
    className="stx-widget-frame stx-widget-frame--scoped the-class"
  >
    <div>
      Hello world!
    </div>
  </div>
</WidgetFrame>
`;
